webui/react/src/stores/experiments.tsx

```diff
@@ -1,24 +1,176 @@
 import { Loadable, Loaded, NotLoaded } from 'hew/utils/loadable';
 import { Map } from 'immutable';
+import * as t from 'io-ts';
 
+import { valueof } from 'ioTypes';
 import { getExperiments } from 'services/api';
-import { V1Pagination } from 'services/api-ts-sdk';
+import { Jobv1State } from 'services/api-ts-sdk';
 import { GetExperimentsParams } from 'services/types';
-import { ExperimentItem, ExperimentPagination } from 'types';
-import { Observable, observable, WritableObservable } from 'utils/observable';
+import {
+  CheckpointStorageType,
+  ExperimentItem,
+  ExperimentPagination,
+  ExperimentSearcherName,
+  HyperparameterBase,
+  Hyperparameters,
+  HyperparameterType,
+  Json,
+  JsonObject,
+  RunState,
+} from 'types';
+import asValueObject, { ValueObjectOf } from 'utils/asValueObject';
+import { immutableObservable, Observable } from 'utils/observable';
 import { encodeParams } from 'utils/store';
 
 import PollingStore from './polling';
 
-type ExperimentCache = {
-  experimentIds: Readonly<number[]>;
-  pagination: Readonly<V1Pagination>;
-};
+// TODO: use these codecs as source for types when possible
+const jsonCodec: t.Type<Json> = t.recursion('Json', () =>
+  t.union([t.string, t.number, t.boolean, t.null, t.array(jsonCodec), jsonObjectCodec]),
+);
+const jsonObjectCodec: t.Type<JsonObject> = t.recursion('JsonObject', () =>
+  t.record(t.string, jsonCodec),
+);
+
+const primitivesCodec = t.union([t.boolean, t.number, t.string]);
+const searcherCodec = t.intersection([
```

Comment:
It feels like a lot of these should be moved to `types.ts`.

Reply:
I don't disagree -- i wanted to replace the existing definitions in types.ts with these as a separate pr because the only consumer of the experiment store (`ClusterOverallStats`) doesn't actually use the experiment type here -- it just uses the pagination type.

Reply:
Can you make a ticket to track that and add it into the sprint then?

Reply:
[done](https://hpe-aiatscale.atlassian.net/browse/WEB-1833)!

Reply:
Thanks.